kobo/apps/openrosa/libs/utils/logger_tools.py

```diff
@@ -833,7 +854,10 @@ def _get_instance(
         instance.xml = xml
         instance._populate_xml_hash()
         instance.uuid = new_uuid
-        instance.save()
+        try:
```

Comment:
I cannot comment on line 845, but I think we want to filter on `root_uuid` instead. 
```python 
if instance := Instance.objects.filter(root_uuid=old_uuid).first():
        check_edit_submission_permissions(request, xform)
        InstanceHistory.objects.create(
            xml=instance.xml, xform_instance=instance, uuid=old_uuid
        )
        instance.xml = xml
        instance._populate_xml_hash()
        instance.uuid = new_uuid
        ...
```